src/ovirtapi/transform.js

```diff
@@ -661,10 +655,79 @@ const StorageDomain = {
   toInternal ({ storageDomain }: { storageDomain: ApiStorageDomainType }): StorageDomainType {
     const permissions = Permissions.toInternal({ permissions: storageDomain?.permissions?.permission })
 
+    // frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#getDiskVolumeFormat
+    const diskAttributesForDiskType: {
+      ['thin' | 'pre']: { sparse: boolean, format?: 'raw' | 'cow' }
+    } = {
+      thin: { sparse: true },
+      pre: { sparse: false },
+    }
+
+    // frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#getVolumeType
+    const defaultDiskFormatToSparse: {
+      ['raw' | 'cow']: boolean
+    } = {
+      cow: true,
+      raw: true,
+    }
+
+    const storageType = storageDomain?.storage?.type ?? 'unknown'
+    let storageSubType
+    switch (storageType) {
+      case 'nfs':
+      case 'localfs':
+      case 'posixfs':
+      case 'glusterfs':
+      case 'glance':
+        storageSubType = 'file'
+        diskAttributesForDiskType.thin.format = 'raw'
+        diskAttributesForDiskType.pre.format = 'raw'
+        defaultDiskFormatToSparse.cow = true
+        defaultDiskFormatToSparse.raw = true
```

Comment:
in Webadmin this value is calculated based on VM compatibility version (see `copyPreallocatedFileBasedDiskSupported`). If simplified approach is OK then please comment on that in the code.

Reply:
More logic to repeat based on config values that are not available via REST!

That logic should probably be added here.  Since it is based on compatibility version / cluster version of the VM, it'll need more handling like config values.

Reply:
I added `isCopyPreallocatedFileBasedDiskSupported` to templates and clusters.  It's calculated in the sagas just like permits/permissions and follows the general structure of the `cpuOptions`.

The only difference is that the base engine config key `CopyPreallocatedFileBasedDiskSupported` is not available to read from the engine.  If we want to do that, we can do it in a future PR.  That would require a few changes on the engine to make it available first.